coordinator/src/services/translation/request/dispense/release.ts

```diff
@@ -5,17 +5,22 @@ import {convertMomentToHl7V3DateTime} from "../../common/dateTime"
 import moment from "moment"
 import pino from "pino"
 import {createAuthorForUnattendedAccess} from "../agent-unattended"
+import {createAuthorForAttendedAccess} from "../agent-attended"
 
 export async function translateReleaseRequest(
   fhirReleaseRequest: fhir.Parameters,
   logger: pino.Logger
 ): Promise<hl7V3.NominatedPrescriptionReleaseRequestWrapper | hl7V3.PatientPrescriptionReleaseRequestWrapper> {
   const organizationParameter = getIdentifierParameterByName(fhirReleaseRequest.parameter, "owner")
   const organizationCode = organizationParameter.valueIdentifier.value
+
+  const authorParameter = getIdentifierParameterOrNullByName(fhirReleaseRequest.parameter, "agent")
+
   const prescriptionIdParameter = getIdentifierParameterOrNullByName(fhirReleaseRequest.parameter, "group-identifier")
-  if (prescriptionIdParameter) {
+  if (prescriptionIdParameter && authorParameter) {
     const prescriptionId = prescriptionIdParameter.valueIdentifier.value
-    return await createPatientReleaseRequest(organizationCode, prescriptionId, logger)
+    const authorProfessionalCode = authorParameter.valueIdentifier.value
+    return await createPatientReleaseRequest(organizationCode, authorProfessionalCode, prescriptionId, logger)
   } else {
     return await createNominatedReleaseRequest(organizationCode, logger)
   }
```

Comment:
is this `if` to make typescript happy or is it possible that we could have `(prescriptionIdParameter == truthy && authorParameter == falsey)` or vice versa?

just wondering if this check here could actually end up losing us a parameter by accident?